providers/universal-provider/src/providers/eip155.ts

```diff
@@ -141,11 +140,38 @@ class Eip155Provider implements IProvider {
     return http;
   }
 
-  private handleSwitchChain(newChainId: string) {
-    const chainId = parseInt(newChainId, 16);
-    const caipChainId = `${this.name}:${chainId}`;
-    validateChainApproval(caipChainId, this.namespace.chains);
-    this.setDefaultChain(`${chainId}`);
+  private async handleSwitchChain(args: RequestParams): Promise<any> {
+    let hexChainId = args.request.params ? args.request.params[0]?.chainId : "0x0";
+    hexChainId = hexChainId.startsWith("0x") ? hexChainId : `0x${hexChainId}`;
+    const parsedChainId = parseInt(hexChainId, 16);
+    // if chainId is already approved, switch locally
+    if (this.isChainApproved(parsedChainId)) {
+      this.setDefaultChain(`${parsedChainId}`);
```

Comment:
If we hit this condition, will metamask somehow know to switch chain in their wallet?

Reply:
Nope, we have to wait for them to actually solve that on their side